src/create/create_request.ts

```diff
@@ -0,0 +1,91 @@
+import {
+  FormattedOracleRequest,
+  OracleRequest,
+  RawOracleRequest,
+} from "../types";
+import { getUTCTimestamp, proofOfWork, formatSubmitRequest } from "../utils";
+/**
+ *
+ * @param method the ethApi method from the request
+ *
+ * @description Currently only eth_call is supported. Upon addition of multiple will change this to an object to attain O(1) lookup time for keys with undefined meaning invalid
+ */
+function _validateEthApiMethod(method: string) {
+  if (method !== "eth_call")
+    throw new Error("Only eth_call is currently supported");
+}
+
+function _formatJsps(jsps: string[]) {
+  return `[${jsps.join(",")}]`;
+}
+
+function _formatTrims(trims: number[]) {
+  return `[${trims.join(",")}]`;
+}
+
+function _addTrims(trims?: number[]) {
+  if (trims) {
+    return `"trims":${_formatTrims(trims)},`;
+  }
+}
+
+function _addPost(post?: string) {
+  if (post) {
+    return `"post":${post.trim()},`;
+  }
+}
+
+function _validateJsps(jsps: string[]) {
+  if (jsps.length === 0) throw new Error("Must have at least one jsps");
+  if (jsps.length >= 32) throw new Error("Exceeded Maximum amount of JSPS");
+}
+
+export default async function createRequest(request: OracleRequest) {
+  _validateJsps(request.jsps);
+  let finalStringRequest: string = "";
+  if (request.type === "raw") {
+    const rawRequest = request as RawOracleRequest;
+    if (rawRequest.rType === "eth") {
+      _validateEthApiMethod(rawRequest.ethApi);
+      finalStringRequest = `{"cid":${rawRequest.cid},"uri":${rawRequest.uri},"ethApi":${rawRequest.ethApi},"params":[{"from":${rawRequest.params.from},"to":${rawRequest.params.to},"data":${rawRequest.params.data},"gas":${rawRequest.params.gas}, "latest"]},"encoding":${rawRequest.encoding},"time":${rawRequest.time},"pow":${rawRequest.pow}}`;
```

Comment:
Maybe let's move bring combination to type function, like RawOracleRequest and FormattedOracleRequest would have a function "toRequestString()" or similar which would return a string of object

Reply:
Interesting - do agree this could be made into more functions but the idea behind the two types is as follows:

1. Using RawOracleRequest is good for tests. Meaning things can be fully hardcoded to achieve full coverage
2. Using RawOracleRequest is good for developers who want 100% control
3. FormattedOracleRequest is perfect for most developers who want Oracle to be handled for them as much as possible by the SDK

Think adding more functions to make this a bit more modular and re-usable is good idea